lib/improver/spotdata_new/apply_lapse_rate.py

```diff
@@ -0,0 +1,206 @@
+                List of cubes for which the attributes should be tested.
+        """
+        # Allow user to bypass cube comparison by setting identifier to None.
+        if self.grid_metadata_identifier is None:
+            return
+
+        reference_attributes = self._get_attributes(cubes[0])
+        for cube in cubes:
+            attributes = self._get_attributes(cube)
+            self._compare_attributes(attributes, reference_attributes)
+
+    def process(self, spot_data_cube, neighbour_cube, gridded_lapse_rate_cube):
+        """
+        Extract lapse rates from the appropriate grid points and apply them to
+        the spot extracted temperatures.
+
+        The calculation is::
+
+         lapse_rate_adjusted_temperatures = temperatures + lapse_rate *
+         vertical_displacement
+
+        Args:
+            spot_data_cube (iris.cube.Cube):
+                The spot data cube of temperatures extracted from the gridded
+                fields. These temperatures will have been extracted using the
+                same neighbour_cube and neighbour_selection_method that are
+                being used here.
+            neighbour_cube (iris.cube.Cube):
+                The neighbour_cube that contains the grid coordinates at which
+                lapse rates should be extracted and the vertical displacement
+                between those grid points on the model orography and the spot
+                data sites actual altitudes.
+            gridded_lapse_rate_cube (iris.cube.Cube):
+                A cube of temperature lapse rates on the same grid as that from
+                which the spot data temperatures were extracted.
+        Returns:
+            new_spot_cube (iris.cube.Cube):
+                A copy of the input spot_data_cube with the data modified by
+                the lapse rates to give a better representation of the site's
+                temperatures.
+        """
+        # Check the cubes are compatible.
+        self.check_grid_match([spot_data_cube, neighbour_cube,
+                               gridded_lapse_rate_cube])
+
+        # Extract the lapse rates that correspond to the spot sites.
+        extraction_plugin = SpotExtraction(
+            neighbour_selection_method=self.neighbour_selection_method)
+        spot_lapse_rate = extraction_plugin.process(neighbour_cube,
+                                                    gridded_lapse_rate_cube)
+
+        # Extract vertical displacements between the model orography and sites.
+        method_constraint = iris.Constraint(
+            neighbour_selection_method_name=self.neighbour_selection_method)
+        data_constraint = iris.Constraint(
+            grid_attributes_key='vertical_displacement')
+        vertical_displacement = neighbour_cube.extract(method_constraint &
+                                                       data_constraint)
+
+        # Create a copy of the input cube with modified temperatures.
```

Comment:
`Apply lapse rate adjustment to temperature at each site`

Reply:
Done.